sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/MetadataReferences/MetadataReferenceFacade.cs

```diff
@@ -0,0 +1,293 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2020 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace SonarAnalyzer.UnitTest.MetadataReferences
+{
+    internal static class MetadataReferenceFacade
+    {
+        internal static IEnumerable<MetadataReference> GetMsCorLib() =>
+#if NETFRAMEWORK
+            FrameworkMetadataReference.Mscorlib;
+#else
+            new[] {CoreMetadataReference.MsCorLib};
+#endif
+
+        internal static IEnumerable<MetadataReference> GetMicrosoftVisualBasic() =>
+#if NETFRAMEWORK
+            FrameworkMetadataReference.MicrosoftVisualBasic;
+#else
+            new[]
+            {
+                CoreMetadataReference.MicrosoftVisualBasic,
+                CoreMetadataReference.MicrosoftVisualBasicCore
+            };
+#endif
+
+        internal static IEnumerable<MetadataReference> GetMicrosoftWin32Registry() =>
+#if NETFRAMEWORK
+            Enumerable.Empty<MetadataReference>();
+#else
+            new[] {CoreMetadataReference.MicrosoftWin32Registry};
+#endif
+        internal static IEnumerable<MetadataReference> GetMicrosoftWin32Primitives() =>
+#if NETFRAMEWORK
+            Enumerable.Empty<MetadataReference>();
+#else
+            new[] {CoreMetadataReference.MicrosoftWin32Primitives};
+#endif
+
+        internal static IEnumerable<MetadataReference> GetSystemCollections() =>
+#if NETFRAMEWORK
+            Enumerable.Empty<MetadataReference>();
```

Comment:
please add comment "is in mscorlib.dll"

Reply:
I'm not sure these comments help (there are 9 places where we return empty). Isn't it obvious that the types are already available?

Reply:
yes, you're actually right